HHS/Head-Start-TTADP
# [no ticket] Don't dedupe on prompts via RTR, plug duplicate goal race condition

## Description of change
Two issues I turned up when working on a support case
- There was duplicative goal information coming from the frontend that created an observable race condition leading to an extra goal being created. I think this is now visible due to code pushed just a few hours ago, so it's unlikely to have created many extra goals.
- The RTR wasn't accounting for FEI goal responses when reducing the goals for display, so it was possible to trap an FEI goal in a non-editable state. 

## How to test
Follow these steps:
- Create a report for a recipient that has two grants, but only select one. 
- Select the FEI goal (the first time for this recipient). Fill out the rest of the form, including the root causes.
- Set yourself as an approver, go back to the activity summary and select the second grant for that same recipient.
- Approve the report, and go to the RTR for that recipient. There should be two entries on the RTR, one for each goal you just created (they aren't condensed because they have different prompt responses.) There should be no extra goals created.

## Issue(s)

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
